backend/tarsy/models/masking_config.py

```
@@ -0,0 +1,121 @@
+            v: The regex pattern string
+            
+        Returns:
+            The validated pattern string
+            
+        Raises:
+            ValueError: If the pattern is not a valid regex
+        """
+        try:
+            re.compile(v)
+        except re.error as e:
+            raise ValueError(f"Invalid regex pattern: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)